### DISCUSSION TITLE
Remove forwardable?

### DISCUSSION THREAD

**Discussion by MSP-Greg** [puma/puma#3190] [General]
[Forwardable](https://github.com/ruby/forwardable) is currently used by server.rb & client.rb.

It is part of the standard library, but it's also available as a gem.  Hence, it could be 'converted' to a bundled gem.  Bundled gems need to be included as dependencies in gemspec files.

Server is delegating four methods of Binder, but Binder is also a available as an `attr_accessor` in Server.  I don't believe any of the delegated methods are used in Puma lib code, but they are used in test code.

Client delegates one method (`closed?`) from its `io`.

Since removing these would be an API change, I think we should remove Forwardable and do so manually.

See https://github.com/ruby/ruby/commit/1e7ddfa7ea

Thoughts?

**Comment by dentarg** [comment]
> Since removing these would be an API change, I think we should remove Forwardable and do so manually.

I agree

**Reply by MSP-Greg** [reply]
@dentarg 

Thanks.   I can do a PR within a few days, or...

**Reply by dentarg** [reply]
Created an issue (https://github.com/puma/puma/issues/3218) so this can be picked up by anyone

**Reply by MSP-Greg** [reply]
@dentarg 

See https://github.com/puma/puma/pull/3191.  Are you okay with the commit/PR title?

**Reply by dentarg** [reply]
Oh damn forgot about that PR 🫣

Yes, message in PR description looks good now, I can squash merge it
